Fix build step

kipparker opened the pull request:
The problem was pip not resolving dependencies correctly, so I've switched to using pipenv which has better dependency resolution and updated the readme. @bensummers could you test if it works ok for you on windows?

kipparker committed:
Add Pipfile for testing by Ben

kipparker committed:
Change instructions to use pipenv

bensummers commented:
On Ubuntu, to avoid an `error: externally-managed-environment` I had to do:

```
sudo apt install python3-pip libffi-dev libjpeg-dev libcairo2-dev
python3 -m venv .venv
. .venv/bin/activate
pip install pipenv
pipenv install
```

But then when I run it, I get:

```
$ mike serve
error: cannot import name '_imagingft' from 'PIL' (/home/ben/ib1-technical-docs-1/.venv/lib/python3.12/site-packages/PIL/__init__.py)
```

kipparker commented:
For ubuntu (because it has lots of python package managed by apt) you'd need to do:
```bash
apt install pipx
pipx install pipenv
```
Then continue. I removed the `pip install pipenv` line and replaced with a link to the Pipenv docs.

kipparker committed:
Remove installation of Pipenv instruction, add a link to the Pipenv docs page
